### PR TITLE
Refactor API calls into adapter

### PR DESCRIPTION
**Pull request by rcdilorenzo**

### CODE REVIEW

File: phc/session.py [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
     """Represents a PHC API session"""
 
+    adapter: Adapter
+
     def __init__(
         self,
         token: str = os.environ.get("PHC_ACCESS_TOKEN"),
         refresh_token: str = os.environ.get("PHC_REFRESH_TOKEN"),
         account: str = os.environ.get("PHC_ACCOUNT"),
+        adapter=Adapter(),
```
Comment (jonathanbranam) [inline phc/session.py:22]: If you create the Adapter as a default argument, then every Session will share the same Adapter instance. Is that OK? That Python behavior is usually a surprise (only evaluating default args once) and I tend to assign `None` as a default arg and then check for None in the code for anything potentially mutable.

FWIW, I'm pretty sure those calls to `os.environ.get` will only be evaluated once also, so if someone creates a `Session`, then updates those env vars and tries to create another `Session`, the updated values will not be used. That might issues I've had trying to use those. I found I had to restart my kernel and set them before importing or calling the SDK for them to be honored.

File: phc/adapter.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+class Adapter:
+    should_refresh: bool
+
+    def __init__(self):
+        self.should_refresh = True
+
+    async def send(
+        self,
+        *,
```
Comment (jonathanbranam) [inline phc/adapter.py:12]: I don't really care, but I'm curious why this is here to force keyword only arguments? Is there a specific concern being addressed?

Comment (rcdilorenzo) [reply phc/adapter.py:12]: No, it was mostly just keeping what was there before. I'm ambivalent as well. 😄

File: phc/session.py (hunk 1 above)
Comment (rcdilorenzo) [reply phc/session.py:22]: Totally true. I can adjust that.

**Review by jonathanbranam — APPROVED** [review]
Looks good. I haven't run the code yet, personally, though.

**Review by joedimarzio — APPROVED** [review]

**Pull request merged**